### PR TITLE
 Support for cloud storage connectors

### PR DESCRIPTION
**Pull request by shreyp135**
#172 
Added cloud storage connectors(sync and async) for aws and gcp.
Added tests for aws and gcp.
Added required env variables in the env template file.

### CODE REVIEW

**Comment by KaranrajM** [comment]
Hi @shreyp135 can you please add the google-cloud-storage and aioboto3 packages to jb-lib's pyproject.toml file and update the respective poetry.lock as well.

File: .env-dev.template [hunk 1]
```diff
@@ -38,6 +38,16 @@ PUBLIC_URL_PREFIX= # Set Tunnel URL if using local storage
...
 # AZURE_STORAGE_CONTAINER=
 
+# Set AWS storage keys if using AWS S3 Storage
+# AWS_S3_BUCKET_NAME=
+# AWS_ACCESS_KEY_ID=
+# AWS_SECRET_ACCESS_KEY=
+# AWS_REGION=
+
+# Set GCP credentials if using GCP Storage
+# GCP_BUCKET_NAME=
```
Comment (KaranrajM) [inline .env-dev.template:48]: Add STORAGE_TYPE=gcp for gcp storage and add STORAGE_TYPE=azure for azure storage in both .env-dev and .env-dev.template files.

**Comment by shreyp135** [comment]
Thanks @KaranrajM for reviewing the pull request.
I have made the necessary changes.

**Comment by KaranrajM** [comment]
Hi @shreyp135 the changes look good. I hope all the test cases are passing.

**Comment by KaranrajM** [comment]
Since this PR was merged, can you push these new changes to a new branch and create a new PR that we can merge it with main when you make these changes? @shreyp135

**Comment by VedantKhairnar** [comment]
Hello @shreyp135   ,
Thanks for your contribution.
As part of Augtoberfest, we want to reward you for your contributions, so pls fill out [this form](https://forms.gle/VBDUPWMYcZEbmVrQ9) by 19th Sept.
Thanks.

**Pull request merged**